library/oneview_power_device.py

```diff
@@ -0,0 +1,272 @@
+        choices: ['present', 'discovered', 'absent', 'power_state_set', 'refresh_state_set', 'uid_state_set']
+        required: true
+    data:
+        description:
+            - List with Power Device properties and its associated states
+        required: true
+notes:
+    - "A sample configuration file for the config parameter can be found at:
+       https://github.com/HewlettPackard/oneview-ansible/blob/master/examples/oneview_config-rename.json"
+'''
+
+EXAMPLES = '''
+- name: Add a Power Device
+  oneview_power_device:
+    config: "{{ config }}"
+    state: present
+    data:
+        name: 'Power Device Name'
+        ratedCapacity: 40
+  delegate_to: localhost
+
+- name: Add an iPDU
+  oneview_power_device:
+    config: "{{ config }}"
+    state: discovered
+    data:
+         hostname : '{{ power_device_hostname }}'
+         username : '{{ power_device_username }}'
+         password : '{{ power_device_password }}'
+         force : false
+  delegate_to: localhost
+
+- name: Power off the Power Device
+  oneview_power_device:
+    config: "{{ config }}"
+    state: power_state_set
+    data:
+        name: 'Power Device Name'
+        powerStateData:
+            powerState: "Off"
+  delegate_to: localhost
+
+- name: Refresh the Power Device
+  oneview_power_device:
+    config: "{{ config }}"
+    state: refresh_state_set
+    data:
+        name: 'Power Device Name'
+        refreshStateData:
+            refreshState : "RefreshPending"
+  delegate_to: localhost
+
+- name: Set UID light state of the Power Device on
+  oneview_power_device:
+    config: "{{ config }}"
+    state: uid_state_set
+    data:
+        name: 'Power Device Name'
+        uidStateData:
+            uidState: "On"
+  delegate_to: localhost
+
+- name: Remove the Power Device by its name
+  oneview_power_device:
+    config: "{{ config }}"
+    state: absent
+    data:
+        name: 'Power Device Name'
+  delegate_to: localhost
+'''
+
+RETURN = '''
+power_device:
+    description: Has the OneView facts about the Power Device.
+    returned: On states 'present', 'discovered', 'power_state_set', 'refresh_state_set', 'uid_state_set'. Can be null.
+    type: complex
+'''
+
+POWER_DEVICE_ADDED = 'Power Device added successfully.'
+POWER_DEVICE_IPDU_ADDED = 'iPDU added successfully.'
+POWER_DEVICE_ALREADY_PRESENT = 'Power Device is already present.'
+POWER_DEVICE_DELETED = 'Power Device deleted successfully.'
+POWER_DEVICE_UPDATED = 'Power Device updated successfully.'
+POWER_DEVICE_ALREADY_ABSENT = 'Power Device is already absent.'
+POWER_DEVICE_NEW_NAME_ALERADY_EXISTS = 'Power Device with new name already exists.'
+POWER_DEVICE_MANDATORY_FIELD_MISSING = "Mandatory field was not informed: data.name"
+POWER_DEVICE_POWER_STATE_UPDATED = 'Power Device power state changed successfully.'
+POWER_DEVICE_REFRESH_STATE_UPDATED = 'Power Device refresh state changed successfully.'
+POWER_DEVICE_UID_STATE_UPDATED = 'Power Device uid state changed successfully.'
```

Comment:
I'm seeing UID being spelled "UID", "UId" and "uid". I think it should be written using only one form to follow a pattern.